[WebGPU] PSO repro generator fails to emit GPU family which is needed for some compilations

mwyrzykowski opened the pull request:
#### c7fd401462442988d579fe1535865433e0759ebf
<pre>
[WebGPU] PSO repro generator fails to emit GPU family which is needed for some compilations
<a href="https://bugs.webkit.org/show_bug.cgi?id=298209">https://bugs.webkit.org/show_bug.cgi?id=298209</a>
radar://159628368

Reviewed by Dan Glastonbury.

Shaders which use the GPU family will fail to compile if we don&apos;t emit
it in the MTLCompileOptions.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::dumpMetalReproCaseComputePSO):
(WebGPU::dumpMetalReproCaseRenderPSO):

Canonical link: <a href="https://commits.webkit.org/299449@main">https://commits.webkit.org/299449@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ad3d119f45307319154d55447a2831c73232bbd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119015 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38696 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29347 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125220 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71078 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/695ca902-38d4-4829-ad2b-6ca47dafafb9) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39392 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47278 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90350 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59820 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d12d05b3-bdbf-4252-848e-f76bcd4bdc5b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121968 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31409 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106701 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70826 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d664af71-2cf9-49ad-b04c-aef684f8c2d4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30472 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68873 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100855 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25003 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128256 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45922 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34699 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98985 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46289 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102921 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98765 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25112 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44225 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22224 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42501 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45792 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51470 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45258 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48604 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46944 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/32a08c21cfeb488b144ec86529485deb3280bd02. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50160

djg commented on Source/WebGPU/WebGPU/Pipeline.mm:
```diff
@@ -408,10 +429,30 @@ void dumpMetalReproCaseRenderPSO(String&& vertexShaderSource, String&& vertexFun
 #import <Foundation/Foundation.h>
 #import <Metal/Metal.h>
 
+    static uint32_t computeAppleGPUFamily(id<MTLDevice> device)
```
Why is this duplicated?

djg reviewed (approved):
:shipit: with nit

mwyrzykowski committed:
[WebGPU] PSO repro generator fails to emit GPU family which is needed for some compilations
https://bugs.webkit.org/show_bug.cgi?id=298209
radar://159628368

Reviewed by Dan Glastonbury.

Shaders which use the GPU family will fail to compile if we don't emit
it in the MTLCompileOptions.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::dumpMetalReproCaseComputePSO):
(WebGPU::dumpMetalReproCaseRenderPSO):

Canonical link: https://commits.webkit.org/299449@main

webkit-commit-queue commented:
Committed 299449@main (c7fd40146244): <https://commits.webkit.org/299449@main>

Reviewed commits have been landed. Closing PR #50160 and removing active labels.
